components/raftstore-v2/src/operation/ready/apply_trace.rs

```diff
@@ -221,6 +225,26 @@ impl ApplyTrace {
         self.admin.flushed
     }
 
+    pub fn should_flush(&mut self) -> bool {
+        if self.admin.flushed != self.admin.last_modified {
+            // It's waiting for other peers, flush will not help.
+            return false;
+        }
+        let last_modified = self.data_cfs.iter().filter_map(|pr| {
+            if pr.last_modified != pr.flushed {
+                Some(pr.last_modified)
+            } else {
+                None
+            }
+        }).max();
+        if let Some(m) = last_modified && m >= self.admin.flushed + 512 && m >= self.last_flush_trigger + 512 {
```

Comment:
A memtable must be flushed after 512 modifications?

Reply:
Yes, otherwise replay can take too much time.

Reply:
I think this number is not reasonable... Even for 4KB rows which is large already, each tablet would flush each 2MB writes, that's tens of flushes per second. And even memtable control is useless anymore, 1024 regions would only take 2GB.

Reply:
An entry is a batch, its maximum size is about 2MiB.

Reply:
I increase it to 4096 for now and see how test works.